File: NewBookStore/wwwroot/js/AddBook.js
// при нажатии на кнопку отправки формы идет запрос к /login для получения токена
document.getElementById("saveBook").addEventListener("click",
    async e => {
        e.preventDefault();

        // отправляет запрос и получаем ответ
        const response = await fetch("/addBook/addBook", {
            method: "POST",
            headers: { "Accept": "application/json", "Content-Type": "application/json" },
            body: JSON.stringify({
                Name: document.getElementById("name").value,
                /*Author: document.getElementById("author").value,*/
                GenreId: document.getElementById("genreId").value,
                PublishingHouse: document.getElementById("publishingHouse").value,
                YearOfPublication: document.getElementById("yearOfPublication").value,
                Description: document.getElementById("description").value
            })
        });
        document.getElementById("name").value = "";
        document.getElementById("author").value = "";
        document.getElementById("genreId").value = "";
        document.getElementById("publishingHouse").value = "";
        document.getElementById("yearOfPublication").value = "";
        document.getElementById("description").value = "";
        window.location.reload();
    });

File: NewBookStore/Models/AddBook.cs
using System.ComponentModel.DataAnnotations;

namespace NewBookStore.Models;

public class AddBook
{
    [Required]
    public required string Name { get; set; }
    [Required]
    public string Author {  get; set; }
    public int GenreId {  get; set; }
    public string? PublishingHouse { get; set; }
    public int YearOfPublication {  get; set; }
    public string? Description { get; set; }

}

